File: src/main/java/es/uji/ei1039/agenda/model/Phone.java
package es.uji.ei1039.agenda.model;

import javafx.beans.property.SimpleStringProperty;
import javafx.beans.property.StringProperty;
import org.jetbrains.annotations.NotNull;

public final class Phone {

    private final @NotNull StringProperty phone = new SimpleStringProperty();

    public String getPhone() {
        return phone.get();
    }

    public @NotNull StringProperty phoneProperty() {
        return phone;
    }

    public void setPhone(final String phone) {
        this.phone.set(phone);
    }


    private final @NotNull StringProperty label = new SimpleStringProperty();

    public String getLabel() {
        return label.get();
    }

    public @NotNull StringProperty labelProperty() {
        return label;
    }

    public void setLabel(final String label) {
        this.label.set(label);
    }
}

File: src/main/kotlin/es/uji/ei1039/agenda/model/Phone.kt
package es.uji.ei1039.agenda.model

import javafx.beans.property.*
import tornadofx.ItemViewModel
import tornadofx.getValue
import tornadofx.setValue

sealed class Phone(
    val id: Int,
    phone: String = "",
    label: Label? = null
) {
    val phoneProperty: StringProperty = SimpleStringProperty(phone)
    var phone: String by phoneProperty

    val labelProperty: ObjectProperty<Label> = SimpleObjectProperty(label)
    var label: Label by labelProperty

    class New : Phone(-1)
    class Existing(
        id: Int,
        phone: String,
        label: Label
    ) : Phone(id, phone, label)

    enum class Label {
        HOME,
        WORK,
        MOBILE
    }

    class ViewModel(phone: Phone) : ItemViewModel<Phone>(phone) {
        val phone: Property<String> = bind(Phone::phoneProperty)
        val label: Property<Label> = bind(Phone::labelProperty)
    }
}
